src/app/components/Application/Main.jsx

```diff
@@ -74,67 +107,85 @@ class Main extends React.Component {
       selectedFilters = this.state.selectedFilters.filter(id => id !== filterId);
     }
 
-    const picks = this.getNewPickSet(this.props.currentPicks.picks, selectedFilters);
+    let picks = this.getNewPickSet(this.props.picksData.picks, selectedFilters);
     const selectableFilters = utils.getSelectableTags(picks);
 
+    picks = this.filterByAudience(picks, this.props.currentAudience, this.props.picksData.type);
+
     this.setState({
       selectableFilters,
       picks,
       selectedFilters,
     });
   }
 
+  /**
+   * getCount(picksData = {})
+   * Returns the count of the current set of picks.
+   * @param {object} picksData
+   */
+  getCount() {
+    const picks = this.state.picks;
+
+    return (picks && picks.length) ? picks.length : 0;
+  }
+
   /**
    * clearFilters()
    * Reset the list of picks and set the list of filters back to its initial state.
-   * @param {object} ref
+   * @param {string} id
    */
-  clearFilters(ref) {
+  clearFilters() {
     const selectedFilters = [];
-    const picks = this.getNewPickSet(this.props.currentPicks.picks, selectedFilters);
+    const picks = this.getNewPickSet(this.props.picksData.picks, selectedFilters);
     const selectableFilters = utils.getSelectableTags(picks);
 
     utils.trackPicks('Clear Filters', 'Clicked');
 
-    if (ref) {
-      ReactDOM.findDOMNode(ref).focus();
-    }
-
     this.setState({
       selectableFilters,
       picks,
       selectedFilters,
     });
+
+    if (document.getElementById('list-title')) {
+      setTimeout(() => {
+        document.getElementById('list-title').focus();
+      }, 400);
+    }
   }
 
   /**
-   * extractAudienceGroup(picks, audience, listType)
+   * filterByAudience(picks, audience, type)
```

Comment:
Updated the function name because it's more of a filter than extracting objects.